### PR TITLE
fix(auth mnesia): fix ctl print error

### PR DESCRIPTION
**Pull request by Rory-Z**
Signed-off-by: zhanghongtong <rory-z@outlook.com>

### CODE REVIEW

**Review by zmstone — COMMENTED** [review]
Why didn’t ~s work? Maybe add a formatter function instead?

**Comment by Rory-Z** [comment]
> Why didn’t ~s work? Maybe add a formatter function instead?

See https://emqx.atlassian.net/browse/EMQX-1078?atlOrigin=eyJpIjoiZmNkMzMzMmY4MzYwNDExYWIwNTM4Y2U1YzU5Y2ZmNWQiLCJwIjoiaiJ9

**Comment by zmstone** [comment]
Try `~ts` ?

**Comment by Rory-Z** [comment]
> Try `~ts` ?

I tried and it still doesn't work

**Review by tigercl — APPROVED** [review]

**Review by zmstone — APPROVED** [review]

File: src/emqx_ctl.erl [hunk 1]
```diff
@@ -153,11 +153,11 @@ usage(CmdParams, Desc) ->
 
 -spec(format(io:format()) -> string()).
 format(Msg) ->
-    lists:flatten(io_lib:format("~s", [Msg])).
+    replace(lists:flatten(io_lib:format("~s", [Msg]))).
```
Comment (zmstone) [inline src/emqx_ctl.erl:156]: wait. change line 144 to `io:format("~s", [format(Format, Args)])` ?

File: src/emqx_ctl.erl [hunk 2]
```diff
@@ -232,3 +232,11 @@ zip_cmd([X | Xs], []) -> [{X, ""} | zip_cmd(Xs, [])];
 zip_cmd([], [Y | Ys]) -> [{"", Y} | zip_cmd([], Ys)];
 zip_cmd([], []) -> [].
 
+replace(Str0) ->
+    case re:run(Str0, "~n$") of
```
Comment (zmstone) [inline src/emqx_ctl.erl:236]: is it to avoid line feed ?
then it should replaced `\n` instead (`\r` too).

Comment (Rory-Z) [reply src/emqx_ctl.erl:236]: I need to replace `~` with `~~`, but at the same time avoid replacing the end of the line with `~n`,

File: src/emqx_ctl.erl (hunk 1 above)
Comment (Rory-Z) [reply src/emqx_ctl.erl:156]: `~s` cannot solve the problem that an error is reported when there is a "~" in the printed content
```
1> io:format("~s", "~").
** exception error: bad argument
     in function  io:format/3
        called as io:format(<0.66.0>,"~s","~")
2> io:format("~").
** exception error: bad argument
     in function  io:format/3
        called as io:format(<0.66.0>,"~",[])
3> io:format("~~").
~ok
```

**Review by zmstone — APPROVED** [review]

**Review by HJianBo — APPROVED** [review]

**Comment by HJianBo** [comment]
Is it necessary to merge it to the master branch?

**Comment by Rory-Z** [comment]
> Is it necessary to merge it to the master branch?

Yep, I'll create another PR later

**Pull request merged**